File: src/Javascript/dynamic programming/119.杨辉三角-ii.js
/*
 * @lc app=leetcode.cn id=119 lang=javascript
 *
 * [119] 杨辉三角 II
 */

// @lc code=start
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
    const level = [];
    // first element
    level.push(1);
    for (let i = 1; i <= rowIndex; i++) {
        for (let j = i - 1; j > 0; j--) {
            level[j] += level[j - 1];
        }
        level.push(1);
    }
    return level;
};
// @lc code=end

File: src/Java/dynamic programming/119.杨辉三角-ii.java
/*
 * @lc app=leetcode.cn id=119 lang=java
 *
 * [119] 杨辉三角 II
 */

// @lc code=start
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> level = new ArrayList<>();
        level.add(1);
        for (int i = 1; i <= rowIndex; i++) {
            for (int j = i - 1; j > 0; j--) {
                level.set(j, level.get(j) + level.get(j - 1));
            }
            level.add(1);
        }
        return level;
    }
}
// @lc code=end

